cartography/intel/azure/key_vaults.py

```diff
@@ -21,59 +19,91 @@
     client = CertificateClient(vault_url, credentials)
     return client
 
+
 @timeit
 def get_key_vaults_client(credentials: Credentials, subscription_id: str) -> KeyVaultManagementClient:
     client = KeyVaultManagementClient(credentials, subscription_id)
     return client
 
 
 @timeit
-def get_key_vaults_list(client: KeyVaultManagementClient, regions: List, common_job_parameters: Dict) -> List[Dict]:
+def get_key_vaults_list(client: KeyVaultManagementClient) -> List[Dict]:
     try:
         key_vaults_list = list(map(lambda x: x.as_dict(), client.vaults.list_by_subscription()))
         return key_vaults_list
     except HttpResponseError as e:
         logger.warning(f"Error while retrieving key vaults - {e}")
         return []
 
+
+def update_access_policies(vault: Dict, common_job_parameters: Dict, client: KeyVaultManagementClient):
+    List_Certificate_Permissions = False
+    for policy in vault.get('properties', {}).get('access_policies', []):
+        if policy.get('object_id') == common_job_parameters['Object_ID'] and 'List' in policy.get('permissions', {}).get('certificates') or 'list' in policy.get('permissions', {}).get('certificates'):
+            List_Certificate_Permissions = True
+            break
+    if not List_Certificate_Permissions:
+        parameters = {
+            "properties": {
+                "access_policies": [
+                    {
+                        "tenant_id": common_job_parameters['AZURE_TENANT_ID'],
+                        "object_id": common_job_parameters['Object_ID'],
+                        "permissions": {
+                            "keys": [
+                                "List",
+                                "Get"
+                            ],
+                            "secrets": [
+                                "List",
+                                "Get"
+                            ],
+                            "certificates": [
+                                "List",
+                                "Get"
+                            ]
+                        }
+                    }
+                ]
+            }
+        }
+        client.vaults.update_access_policy(resource_group_name=vault['resource_group'], vault_name=vault['name'], operation_kind='add', parameters=parameters)
+
+
 @timeit
-def transform_key_vaults(key_vaults: List[Dict], regions: List, common_job_parameters: Dict) -> List[Dict]:
+def transform_key_vaults(client: KeyVaultManagementClient, key_vaults: List[Dict], regions: List, common_job_parameters: Dict) -> List[Dict]:
     key_vaults_data = []
     for vault in key_vaults:
         x = vault['id'].split('/')
         vault['resource_group'] = x[x.index('resourceGroups') + 1]
         vault['consolelink'] = azure_console_link.get_console_link(
             id=vault['id'], primary_ad_domain_name=common_job_parameters['Azure_Primary_AD_Domain_Name'])
+        update_access_policies(vault, common_job_parameters, client)
```

Comment:
@ganeshjagdale9545 we should have exception handling here. So that even if this step fails, our key vault ingestion or cartography process does not crash. can you add try/catch here?